numba/core/pythonapi.py

```diff
@@ -134,7 +134,13 @@ def read_const(self, index):
         """
         assert index < len(self.env.consts)
 
-        return self.pyapi.list_getitem(self.env_body.consts, index)
+        builder = self.pyapi.builder
+        consts = self.env_body.consts
+        ret = cgutils.alloca_once(builder, self.pyapi.pyobj, zfill=True)
+        with builder.if_then(cgutils.is_not_null(builder, consts)):
+            getitem = self.pyapi.list_getitem(consts, index)
```

Comment:
this is unchecked, whilst unlikely, could raise e.g. IndexError?

Reply:
In that case `getitem` would be `NULL` and would be returned to the caller. The caller can then check for the NULL return. There isn't much this function can do.

Caller are handling it like:
- https://github.com/sklam/numba/blob/0d8266613e568b344c13a8d2ee3937e9510d1b49/numba/core/pylowering.py#L425-L426
- https://github.com/numba/numba/pull/6141/files/0d8266613e568b344c13a8d2ee3937e9510d1b49#diff-1e3a0be1e38fdb57e5bbff6f233dd1da
- I will make sure `numba_recreate_record` is checking the `dtype` argument

Reply:
> I will make sure numba_recreate_record is checking the dtype argument

Did ^ get checked?

Reply:
yes

Reply:
thanks !